e2e-polybft/e2e/governance_test.go

```diff
@@ -0,0 +1,330 @@
+package e2e
+
+import (
+	"fmt"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/0xPolygon/polygon-edge/command/sidechain"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft"
+	"github.com/0xPolygon/polygon-edge/consensus/polybft/contractsapi"
+	"github.com/0xPolygon/polygon-edge/contracts"
+	"github.com/0xPolygon/polygon-edge/crypto"
+	"github.com/0xPolygon/polygon-edge/e2e-polybft/framework"
+	"github.com/0xPolygon/polygon-edge/txrelayer"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/stretchr/testify/require"
+	"github.com/umbracle/ethgo"
+)
+
+type VoteType uint8
+type ProposalState uint8
+
+const (
+	Against VoteType = iota
+	For
+	Abstain
+)
+
+const (
+	Pending ProposalState = iota
+	Active
+	Canceled
+	Defeated
+	Succeeded
+	Queued
+	Expired
+	Executed
+)
+
+func TestE2E_Governance_ProposeAndExecuteSimpleProposal(t *testing.T) {
+	var (
+		oldEpochSize = uint64(10)
+		newEpochSize = uint64(20)
+		votingPeriod = 2 * oldEpochSize
+	)
+
+	cluster := framework.NewTestCluster(t, 5, framework.WithEpochSize(int(oldEpochSize)),
+		framework.WithGovernanceVotingPeriod(votingPeriod),
+		framework.WithGovernanceVotingDelay(1))
+	defer cluster.Stop()
+
+	cluster.WaitForReady(t)
+
+	proposer := cluster.Servers[0] // first validator is governor admin by default in e2e tests
+
+	proposerAcc, err := sidechain.GetAccountFromDir(proposer.DataDir())
+	require.NoError(t, err)
+
+	l2Relayer, err := txrelayer.NewTxRelayer(txrelayer.WithClient(proposer.JSONRPC()))
+	require.NoError(t, err)
+
+	t.Run("successful change of epoch size", func(t *testing.T) {
+		// propose a new epoch size
+		setNewEpochSizeFn := &contractsapi.SetNewEpochSizeNetworkParamsFn{
+			NewEpochSize: big.NewInt(int64(newEpochSize)),
+		}
+
+		proposalInput, err := setNewEpochSizeFn.EncodeAbi()
+		require.NoError(t, err)
+
+		proposalDescription := fmt.Sprintf("Change epoch size from %d to %d", oldEpochSize, newEpochSize)
+
+		proposalID := sendProposalTransaction(t, l2Relayer, proposerAcc.Ecdsa,
+			proposalInput, proposalDescription)
+
+		fmt.Println(proposalID)
```

Comment:
Is this a leftover and if it isn't, why not use `t.Log`?
```suggestion
		t.Log("Proposal ID", proposalID)
```

Reply:
Yep. I'll remove it.

Reply:
Done: https://github.com/0xPolygon/polygon-edge/pull/1761/commits/2b42814c7d23f68e9cdcc252c6b71e30e1c2fe76